pkg/cloudprovider/cloudprovider.go

```diff
@@ -1,47 +1,124 @@
 package cloudprovider
 
 import (
+	"encoding/json"
 	"io"
+	"io/ioutil"
 
+	"github.com/k3s-io/k3s/pkg/util"
 	"github.com/k3s-io/k3s/pkg/version"
-	"k8s.io/client-go/informers"
-	informercorev1 "k8s.io/client-go/informers/core/v1"
-	"k8s.io/client-go/tools/cache"
+	"github.com/rancher/wrangler/pkg/apply"
+	"github.com/rancher/wrangler/pkg/generated/controllers/apps"
+	appsclient "github.com/rancher/wrangler/pkg/generated/controllers/apps/v1"
+	"github.com/rancher/wrangler/pkg/generated/controllers/core"
+	coreclient "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/rancher/wrangler/pkg/generic"
+	"github.com/rancher/wrangler/pkg/start"
+	"github.com/sirupsen/logrus"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
 	cloudprovider "k8s.io/cloud-provider"
 )
 
+// Config describes externally-configurable cloud provider configuration.
+// This is normally unmarshalled from a JSON config file.
+type Config struct {
+	LBEnabled   bool   `json:"lbEnabled"`
+	LBImage     string `json:"lbImage"`
+	LBNamespace string `json:"lbNamespace"`
+	Rootless    bool   `json:"rootless"`
+}
+
 type k3s struct {
-	nodeInformer          informercorev1.NodeInformer
-	nodeInformerHasSynced cache.InformerSynced
+	Config
+
+	client   kubernetes.Interface
+	recorder record.EventRecorder
+
+	processor      apply.Apply
+	daemonsetCache appsclient.DaemonSetCache
+	nodeCache      coreclient.NodeCache
+	podCache       coreclient.PodCache
+	workqueue      workqueue.RateLimitingInterface
 }
 
 var _ cloudprovider.Interface = &k3s{}
-var _ cloudprovider.InformerUser = &k3s{}
 
 func init() {
 	cloudprovider.RegisterCloudProvider(version.Program, func(config io.Reader) (cloudprovider.Interface, error) {
-		return &k3s{}, nil
+		var err error
+		k := k3s{
+			Config: Config{
+				LBEnabled:   true,
```

Comment:
Probably stupid but I just wanted to check something. Seeing here a hardcoded true, makes me wonder if LBEnabled stays always true? I see that we are writing `serverConfig.ControlConfig.DisableServiceLB` in a file and then passing that file path to `ccm.InitCloudProvider` but I am not sure what happens next and if that config gets propagated to the k3s cloud provider object

Reply:
These are defaults for when no config file is passed. If the config reader is non-nil, the defaults here will be wiped out by the Unmarshal call.